crates/evm/src/system_calls/eip7251.rs

```
@@ -14,19 +14,10 @@ use revm_primitives::{ExecutionResult, ResultAndState};
 ///
 /// Note: this does not commit the state changes to the database, it only transact the call.
 #[inline]
-pub(crate) fn transact_consolidation_requests_contract_call<EvmConfig, EXT, DB>(
-    evm_config: &EvmConfig,
-    evm: &mut Evm<'_, EXT, DB>,
-) -> Result<ResultAndState, BlockExecutionError>
-where
-    DB: Database,
-    DB::Error: core::fmt::Display,
-    EvmConfig: ConfigureEvm,
-{
-    // get previous env
-    let previous_env = Box::new(evm.context.env().clone());
-
-    // Fill transaction environment with the EIP-7251 consolidation requests contract message data.
+pub(crate) fn transact_consolidation_requests_contract_call(
+    evm: &mut impl Evm<Error: Display>,
```

Should we hide the Error type as well? Seems off to have this generic everywhere

---

we could just add `Error` bound to the trait definition, don't think there's a nice a way to hide a generic error when consuming

---

actually it seems that doing `trait Evm { Error: Display }` just results in the need to enforcing this bound on database, i.e we'd need `DB: Database<Error: Display>` everywhere

I think we can keep this as is for now, and then maybe hide it behind an additional layer of traits on alloy-evm side?